overlord/state/taskrunner.go

```diff
@@ -85,6 +88,11 @@ func NewTaskRunner(s *State) *TaskRunner {
 	}
 }
 
+// SetTaskErrorCallback sett error callback to execute when task errors out.
+func (r *TaskRunner) SetTaskErrorCallback(f func(err error)) {
```

Comment:
we don't use Callback that often explicity,  I would call this (and then related names) along SetOnTaskError, or even just OnTaskError

Reply:
+1 for OnTaskError.